ivy_tests/test_ivy/test_frontends/test_tensorflow/test_general_functions.py

```diff
@@ -267,6 +267,58 @@ def test_tensorflow_eye(
     )
 
 
+@st.composite
+def elems_reshaped(draw):
+    shape = draw(
+        st.lists(st.integers(min_value=1, max_value=10), min_size=1, max_size=4)
+    )
+    elems_size = np.prod(shape).item()
+    elems = draw(
+        st.lists(
+            st.floats(min_value=-1000, max_value=1000),
+            min_size=elems_size,
+            max_size=elems_size,
+        )
+    )
+    return np.reshape(elems, shape)
+
+
+# foldl
+@handle_frontend_test(
+    fn_tree="tensorflow.foldl",
+    fn=st.sampled_from(
+        [
+            lambda a, b: a + b,
+            lambda a, b: a - b,
+            lambda a, b: a * b,
+        ],
+    ),
+    initializer=st.one_of(st.none(), st.floats(min_value=-1000, max_value=1000)),
+    dtype=helpers.get_dtypes("float", full=False),
+    elems_reshaped=elems_reshaped(),
+)
+def test_tensorflow_foldl(
```

Comment:
Try to test for other kwargs aside from `name` as well

Reply:
A side note on a remark previously made by you regarding having all the same parameters to the function as `tf.foldl`; I removed the `back_prop` arg as it is deprecated, and will be removed in coming version of tensorflow. Instead using `ivy.stop_gradient` would be the way to go.

Reply:
> I am not sure what you meant by this, I tried looking for examples of how certain test functions or other function implementations deal with the name argument but wasn't able to arrive at a clearer picture.

Yeah don't test for `name`, but please include `parallel_iterations` and `swap_memory` even if they don't do anything for now. We may update the frontend to make use of them in the future and forget that the test didn't account for them.

Reply:
Consider it done, will update asap 🤝.